File: backend/src/main/java/com/backend/repository/MerchantRepository.java
package com.backend.repository;

import java.util.Optional;
import java.util.UUID;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import com.backend.model.Merchant;

@Repository
public interface MerchantRepository extends CrudRepository<Merchant, UUID> {
  Optional<Merchant> findByUser_UserId(UUID userId);
}

File: frontend/app/src/main/java/com/example/frontendinternship/domain/repository/MerchantRepository.kt
package com.example.frontendinternship.domain.repository

import com.example.frontendinternship.domain.model.MerchantModel
import com.example.frontendinternship.domain.model.ProductModel
import com.example.frontendinternship.domain.model.ShopModel
import java.util.UUID


interface MerchantRepository {
    suspend fun getCurrentMerchantID(): UUID

    suspend fun loadMerchants(): List<MerchantModel>
    suspend fun updateMerchant(userId: UUID, merchantModel: MerchantModel)
}